text/0095-nft-standard.md

```diff
@@ -0,0 +1,371 @@
+
+    /// Transfers the ownership of an NFT without checking the recipient contract.
+    ///
+    /// @param from - The current owner of the NFT.
+    /// @param to - The new owner.
+    /// @param token_id - The NFT ID.
+    fn transfer_from(&mut self, from: Key, to: Key, token_id: U256);
+
+    /// Approves another account or contract to transfer a specific NFT.
+    ///
+    /// @param to - The account that will be granted approval.
+    /// @param token_id - The NFT ID.
+    fn approve(&mut self, spender: Key, token_id: U256);
+
+    /// Revokes approval for a specific NFT.
+    ///
+    /// @param token_id - The NFT ID.
+    fn revoke_approval(&mut self, token_id: U256);
+
+    /// Gets the approved account or contract for a specific NFT.
+    ///
+    /// @param token_id - The NFT ID.
+    /// @return Option<Key> - Approved spender account if one exists.
+    fn get_approved(&self, token_id: U256) -> Option<Key>;
+
+    /// Enables operator approval for all of the caller's NFTs.
+    ///
+    /// @param operator - The operator address to be approved.
+    fn approve_for_all(&mut self, operator: Key);
+
+    /// Revokes operator approval for all of the caller's NFTs.
+    ///
+    /// @param operator - The operator address whose approval is to be revoked.
+    fn revoke_approval_for_all(&mut self, operator: Key);
+
+    /// Checks if an operator is approved to manage all NFTs of the owner.
+    ///
+    /// @param owner - The NFT owner's address.
+    /// @param operator - The operator to check.
+    /// @return bool - True if the operator is approved for all NFTs, false otherwise.
+    fn is_approved_for_all(&self, owner: Key, operator: Key) -> bool;
+}
+```
+
+### Recommended Adoption of the Casper On-chain Contract Metadata Standard CEP96
+
+To improve the integrity, discoverability, and user experience within the Casper ecosystem, implementers of the CEP95 NFT Standard are strongly encouraged to also adopt the **CEP96 Casper On-chain Contract Metadata Standard**.
+
+This standard provides a clear, unified way for contracts to publish immutable, verifiable metadata about themselves directly on-chain, including:
+
+- **Contract Name** (`name`)
+- **Description** (`description`)
+- **Icon URI** (`icon_uri`)
+- **Project URI** (`project_uri`)
+
+By implementing these metadata fields, CEP95 contracts become easily identifiable, understandable, and trustworthy to end-users, explorers, wallets, and marketplaces. The adoption of the On-chain Contract Metadata Standard enhances the overall integrity and usability of the Casper blockchain ecosystem and helps build user trust.
+
+We strongly recommend incorporating this metadata standard alongside CEP95 implementations to contribute positively to ecosystem consistency, transparency, and user-friendliness.
+
+### Recommended Storage for contract data (symbol, etc.)
+
+According to the CEP96 standard, it is recommended to store contract metadata as a form of contract level named keys using `Urefs`.
```

Comment:
`According to the CEP96 standard` is not needed here since we are looking at the standard iteself